From: Coly Li <coyli@suse.de>
Date: Mon, 17 Nov 2008 12:38:22 +0800
Subject: ocfs2: fix return value set in init_dlmfs_fs()

In init_dlmfs_fs(), if calling kmem_cache_create() failed, the code will use return value from
calling bdi_init(). The correct behavior should be set status as -ENOMEM before going to "bail:".

Signed-off-by: Coly Li <coyli@suse.de>
Acked-by: Sunil Mushran <sunil.mushran@oracle.com>
Signed-off-by: Mark Fasheh <mfasheh@suse.com>
---
 fs/ocfs2/dlm/dlmfs.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

Index: linux-2.6.27-ocfs2/fs/ocfs2/dlm/dlmfs.c
===================================================================
--- linux-2.6.27-ocfs2.orig/fs/ocfs2/dlm/dlmfs.c
+++ linux-2.6.27-ocfs2/fs/ocfs2/dlm/dlmfs.c
@@ -608,8 +608,10 @@ static int __init init_dlmfs_fs(void)
 				0, (SLAB_HWCACHE_ALIGN|SLAB_RECLAIM_ACCOUNT|
 					SLAB_MEM_SPREAD),
 				dlmfs_init_once);
-	if (!dlmfs_inode_cache)
+	if (!dlmfs_inode_cache) {
+		status = -ENOMEM;
 		goto bail;
+	}
 	cleanup_inode = 1;
 
 	user_dlm_worker = create_singlethread_workqueue("user_dlm");
